### DISCUSSION TITLE
Does it only check for updates when I run it?

### DISCUSSION THREAD

**Discussion by HubKing** [mjakeman/extension-manager#394] [Q&A] [answered]
It seems that when I start E.M., it checks for updates, and if updates are available and I need to log out to install them. Is that the only time it checks for updates? Can't it check for updates along with the system's update (e.g., pacman on Arch), or can't it automatically install updated extensions when I log out next time? Or, can't it install updated extensions without actually logging out an re-logging in (by typing the password) but simply by restarting the Gnome session (all GUI applications will be terminated, but at least I won't need to type the password again)?

**Comment by Francewhoa** [comment, marked as answer]
Hello @HubKing :) I have the same question. I searched for an answer and found this from the maintainer: 

------------------

#### [Extension Manager registers] with GNOME Shell and it handles all extension downloads/updating/etc. Whenever the user logs in [GNOME], Shell will automatically check for updates.

* Source 1 at https://github.com/mjakeman/extension-manager/issues/346#issuecomment-1465454504
* Source 2 https://github.com/mjakeman/extension-manager/issues/120#issue-1169960743

------------------

#### GNOME Shell does not support updating individual extensions
   
* Source 1 https://github.com/mjakeman/extension-manager/issues/203#issuecomment-1149242796
* Source 2 https://github.com/mjakeman/extension-manager/issues/189#issuecomment-1149228667

**Reply by mjakeman** [reply]
Yep, this is pretty much correct^

GNOME Shell handles all updates, and does this automatically for you. When you log in, GNOME Shell detects if either GNOME Extensions or Extension Manager are installed and then:

 1. Checks for updates
 2. Shows a pop-up notification
 3. Installs them whenever you next log-in

There's one caveat where this doesn't work for certain versions of GNOME 42 and below (workaround: https://github.com/mjakeman/extension-manager/wiki/Known-Issue:-Updates)

**Reply by Francewhoa** [reply]
Thanks for your reply @mjakeman :)

This is to confirm that both options successfully worked. I mean either updating using the log-in GNOME or the workaround.

* Using the latest GNOME 43.9 on Debian 12 Bookworm with Wayland, we tried the log-in GNOME option. It works. This is great.

* Using outdated an GNOME 3.30 on Debian 10 Buster, X11, on a virtual machine, we tried the workaround option. It works too.

For most newcomers, the log-in GNOME option is likely to be more stable. Because, as you know, it is like a fresh start for both all extensions and apps using extensions. Thus, more stable. Also, on the latest Debian 12, Wayland is the default for GNOME log-in. And Wayland presently does not support restarting GNOME during the GNOME session. Users need to log-out then log-in GNOME to restart GNOME. This challenge is resolved by using the log-in GNOME option. While at the same time, advanced end-users are able to use the workaround option. Assuming they use it appropriately.
